### DISCUSSION TITLE
Could I back up vector indexes by back up S3 and Etcd snapshot?

### DISCUSSION THREAD

**Discussion by Pieerot** [milvus-io/milvus#29277] [Q&A and General discussion]
I have used the milvus-backup tool for data backup and recovery. However, the vector index needs to be re-created after data recovery, which will take a long time. So I want to try to back up the snapshot of etcd myself, and at the same time fully back up the S3 data to other buckets. When performing data recovery, I can directly connect to the backed-up S3 and use snapshot to restore the metadata of etcd. Is this method feasible?

**Comment by yhmo** [comment]
Theoretically, this approach can work. Stop the server before you back up the etcd and s3.

**Reply by Pieerot** [reply]
thanks for ansewer. but i can't stop my server when backup. So are there any risks in this process?

**Reply by yhmo** [reply]
During runtime, if there is data inserted, or some tasks(index tasks, compaction tasks, etc) are not finished, milvus will continually write metadata into etcd. If you copy the etcd and s3 at different times, they might be inconsistent, and you will get errors or potential risks when you use the inconsistent etcd/s3 to launch another milvus server.

**Comment by shrawani92** [comment]
could you please post your milvus.yaml config? Whats the s3 end point address?

**Reply by yhmo** [reply]
I think the milvus.yaml is standard. s3 endpoint is private.

**Comment by xiaofan-luan** [comment]
backup index won't be helpful. without original data and it's meta information, there is no way we can recreate the cluster.

Unless you backup the entire S3 bucket and etcd
